ibm_db.c

```diff
@@ -1438,19 +1438,19 @@ static PyObject *_python_ibm_db_connect_helper( PyObject *self, PyObject *args,
 #ifdef __MVS__
                 rc = SQLConnectW((SQLHDBC)conn_res->hdbc,
                     database,
-                    PyUnicode_GetSize(databaseObj)*2,
+                    PyUnicode_GetLength(databaseObj)*2,
                     uid,
-                    PyUnicode_GetSize(uidObj)*2,
+                    PyUnicode_GetLength(uidObj)*2,
                     password,
-                    PyUnicode_GetSize(passwordObj)*2);
+                    PyUnicode_GetLength(passwordObj)*2);
 #else
                 rc = SQLConnectW((SQLHDBC)conn_res->hdbc,
                     database,
-                    PyUnicode_GetSize(databaseObj),
```

Comment:
@bchoudhary6415 Previously we size of uidObj was passed to SQLonnectW, not you are passing length. It may cause data loss. Should we not add 1 to the length to get size?

Reply:
Wherever the size is changed to length function, we need to do +1 to get size.

Reply:
@bimalkjha As we discussed internally, it's perfectly fine to use PyUnicode_GetLength() instead of PyUnicode_GetSize().
Also we can refer the link for the replacement of Deprecated C API https://docs.python.org/3.10/whatsnew/3.3.html?highlight#deprecated-functions-and-types-of-the-c-api